include/nanorange/views/subrange.hpp

```diff
@@ -245,12 +245,15 @@ class subrange : public view_interface<subrange<I, S, K>> {
 
     template <subrange_kind KK = K>
     constexpr auto size() const
-        -> std::enable_if_t<KK == subrange_kind::sized, iter_difference_t<I>>
+        -> std::enable_if_t<KK == subrange_kind::sized, std::make_unsigned_t<iter_difference_t<I>>>
     {
         if constexpr (StoreSize) {
             return data_.size_;
         } else {
-            return data_.end_ - data_.begin_;
+            constexpr auto make_unsigned_like = [](auto i) {
```

Comment:
This lambda was copied over from `iota.hpp`. Maybe we should extract it into some common header?